src/quafu/circuits/quantum_circuit.py

```diff
@@ -267,142 +267,8 @@ def from_openqasm(self, openqasm: str):
-                                    self.x(inds[0])
-                                elif gatename == "y":
-                                    self.y(inds[0])
-                                elif gatename == "z":
-                                    self.z(inds[0])
-                                elif gatename == "h":
-                                    self.h(inds[0])
-                                elif gatename == "id":
-                                    self.id(inds[0])
-                                elif gatename == "s":
-                                    self.s(inds[0])
-                                elif gatename == "sdg":
-                                    self.sdg(inds[0])
-                                elif gatename == "t":
-                                    self.t(inds[0])
-                                elif gatename == "tdg":
-                                    self.tdg(inds[0])
-                                elif gatename == "sx":
-                                    self.sx(inds[0])
-                                elif gatename == "ccx":
-                                    self.toffoli(inds[0], inds[1], inds[2])
-                                elif gatename == "cswap":
-                                    self.fredkin(inds[0], inds[1], inds[2])
-                                elif gatename == "u1":
-                                    self.rz(inds[0], paras[0])
-                                elif gatename == "u2":
-                                    self.rz(inds[0], paras[1])
-                                    self.ry(inds[0], pi / 2)
-                                    self.rz(inds[0], paras[0])
-                                elif gatename == "u3":
-                                    self.rz(inds[0], paras[2])
-                                    self.ry(inds[0], paras[0])
-                                    self.rz(inds[0], paras[1])
-                                elif gatename == "rxx":
-                                    self.rxx(inds[0], inds[1], paras[0])
-                                elif gatename == "ryy":
-                                    self.ryy(inds[0], inds[1], paras[0])
-                                elif gatename == "rzz":
-                                    self.rzz(inds[0], inds[1], paras[0])
-                                else:
-                                    print(
-                                        "Warning: Operations %s may be not supported by QuantumCircuit class currently."
-                                        % gatename
-                                    )
-
-        if not self.measures:
-            self.measures = dict(zip(range(self.num), range(self.num)))
-        if not global_valid:
-            print(
-                "Warning: All operations after measurement will be removed for executing on experiment"
-            )
+        from quafu.qfasm.qfasm_convertor import qasm2_to_quafu_qc
```

Comment:
[import-outside-toplevel](https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/import-outside-toplevel.html) is not recommended by pylint.

This [post](https://stackoverflow.com/questions/1699108/local-import-statements-in-python) why we rarely import modules inside a function.

I would suggest you consider giving up this style in the future ^_^